src/neo/SmartContract/Native/NativeContract.cs

```diff
@@ -28,6 +29,7 @@ public abstract class NativeContract
         public uint ServiceHash { get; }
         public byte[] Script { get; }
         public UInt160 Hash { get; }
+        public int Id { get; } = nextId--;
```

Comment:
Is this deterministic? I think that if we add a new class we can obtain a different value for the old native contracts.

Reply:
https://github.com/neo-project/neo/blob/35c73cecf6538f70005fab1567367a6384b24092/src/neo/SmartContract/Native/NativeContract.cs#L22-L25

This order is deterministic.

Reply:
I think that it's more clear if we pass the Id in the constructor, otherwise we need to remember that this order it's important

Reply:
Could we use symbol+addr？eg：NEX-AJpe4HnhmhAv4iFMy9Gg2Wd3ouBBExi1Lx

Reply:
Reverted to abstract Id.